### PR TITLE
d-SEAMS,rang: Init at 1.0.1, 3.0.1

### PR DESCRIPTION
**Pull request by HaoZeke**
###### Motivation for this change
Initialized
- [d-SEAMS](https://dseams.info)
- [rang](https://github.com/agauniyal/rang/)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) **N/A**
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/chemistry/d-seams/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ clangStdenv, stdenv, fetchurl, catch2, rang, fmt, libyamlcpp, cmake, eigen
+, lua, luaPackages, liblapack, blas, lib, boost, gsl ? null }:
```
Comment (markuskowa) [inline pkgs/applications/science/chemistry/d-seams/default.nix:2]: gsl is not a big library. Would it make sense to just always build with gsl support?

File: pkgs/applications/science/chemistry/d-seams/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+clangStdenv.mkDerivation rec {
+  version = "v1.0.1";
+  pname = "d-SEAMS";
+
+  src = fetchurl {
```
Comment (markuskowa) [inline pkgs/applications/science/chemistry/d-seams/default.nix:8]: Can we use 'fetchFromGitHub' here instead?

File: pkgs/applications/science/chemistry/d-seams/default.nix (hunk 1 above)
Comment (HaoZeke) [reply pkgs/applications/science/chemistry/d-seams/default.nix]: Yup, excellent point, done ^_^

File: pkgs/applications/science/chemistry/d-seams/default.nix (hunk 2 above)
Comment (HaoZeke) [reply pkgs/applications/science/chemistry/d-seams/default.nix]: Sure, have switched to it now. Good catch!

**Comment by markuskowa** [comment]
@GrahamcOfBorg build d-seams

**Pull request merged**